uploaders/common_test.go

```diff
@@ -0,0 +1,138 @@
+package uploaders
+
+import (
+	"image"
+	"image/png"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+func Test_uploadArtifact(t *testing.T) {
+	const contentType = "image/png"
+
+	file, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Errorf("setup: failed to create file, error: %s", err)
```

Comment:
Note
I would use `t.Fatalf` for the initialization part of the test.
E.g: It is unnecessary to run the test cases ff it failed to open/ read the file.